### ISSUE TITLE
Custom metrics not displaying correctly, print accuracy value instead

### ISSUE THREAD

**Issue by cathey** [rstudio/keras#345]
Hi,

I'm working with an imbalanced dataset, so I wrote my own metrics to evaluate the performance (just precision, recall, & f1-score). Here's the code:

![image](https://user-images.githubusercontent.com/9155384/38297813-28f4f8cc-3828-11e8-8581-beac9c06f0ba.png)

When I run the code, precision, recall & f1-score are all printed incorrectly. After some analysis, I found that the categorical accuracy is printed instead of these metrics:

![image](https://user-images.githubusercontent.com/9155384/38297951-806d42ee-3828-11e8-90d1-3b526cbdc565.png)

I'm running Keras 2.1.5 and Tensorflow 1.6.0 on Python 3.6

Can somebody tell me what's going on? Thanks.

**Comment by jjallaire** [comment]
This looks like an issue for the Python version of Keras so you should file it on the Python repo here: https://github.com/keras-team/keras/issues
